contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L2DepositedERC20.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.8.0;
+pragma experimental ABIEncoderV2;
+
+/* Interface Imports */
+import { iOVM_L2DepositedERC20 } from "../../../iOVM/bridge/tokens/iOVM_L2DepositedERC20.sol";
+import { iOVM_L1ERC20Gateway } from "../../../iOVM/bridge/tokens/iOVM_L1ERC20Gateway.sol";
+
+/* Contract Imports */
+import { UniswapV2ERC20 } from "../../../libraries/standards/UniswapV2ERC20.sol";
+
+/* Library Imports */
+import { OVM_CrossDomainEnabled } from "../../../libraries/bridge/OVM_CrossDomainEnabled.sol";
+
+/**
+ * @title OVM_L2DepositedERC20
+ * @dev The L2 Deposited ERC20 is an ERC20 implementation which represents L1 assets deposited into L2.
+ * This contract mints new tokens when it hears about deposits into the L1 ERC20 gateway.
+ * This contract also burns the tokens intended for withdrawal, informing the L1 gateway to release L1 funds.
+ *
+ * Compiler used: optimistic-solc
+ * Runtime target: OVM
+ */
+contract OVM_L2DepositedERC20 is iOVM_L2DepositedERC20, UniswapV2ERC20, OVM_CrossDomainEnabled {
+    
+    /******************
+     * Contract Events*
+     ******************/
+
+    event Initialized(iOVM_L1ERC20Gateway _l1ERC20Gateway);
+
+    /********************************
+     * External Contract References *
+     ********************************/
+
+    iOVM_L1ERC20Gateway l1ERC20Gateway;
+
+    /********************************
+     * Constructor & Initialization *
+     ********************************/
+
+    /**
+     * @param _l2CrossDomainMessenger L1 Messenger address being used for cross-chain communications.
+     * @param _decimals L2 ERC20 decimals
+     * @param _name L2 ERC20 name
+     * @param _symbol L2 ERC20 symbol
+     */
+    constructor(
+        address _l2CrossDomainMessenger,
+        uint8 _decimals,
+        string memory _name,
+        string memory _symbol
+    )
+        public
+        OVM_CrossDomainEnabled(_l2CrossDomainMessenger)
+        UniswapV2ERC20(_decimals, _name, _symbol)
+    {}
+
+    // Should we be inheriting this initialization logic?
```

Comment:
Where would you inherit from? Are you referring to `OVM_CrossDomainEnabled` ?

Reply:
Ah or you're talking about using some standard `InitializableContract` thing?

If that's the case IMO don't worry about inheriting it. I like the clarity of having everything in one spot so you're not tracking down various inherited contracts

Reply:
That is what I was referring to--sounds good.